api/comment_threads.rb

```diff
@@ -88,6 +88,12 @@
 end
 
 delete "#{APIPREFIX}/threads/:thread_id" do |thread_id|
-  thread.destroy
+  begin
+    thread.destroy
+  rescue Elasticsearch::Transport::Transport::Errors::NotFound
+    # If the thread is not in the index, that's actually a good thing given that we just removed it.
+    # Note that this exception will probably only be encountered for tests that don't wait for the index
+    # to be refreshed before attempting to destroy a newly-recreated thread.
```

Comment:
I hope we do not have any tests that will encounter this case. Or should there exist one?